File: loadAseqsJson.py
#!/usr/bin/python3
import os
import sys
from pymongo import MongoClient
import json
import datetime
from Common import *


usage = """
Usage: $0 <database json>

  <database json> should consist of an entire json object per line

  Loads into the aseqs collection.

	-id "TheId" cross reference identifier: gi uni ENS
"""

g_File = sys.argv[1]

if '-col' in sys.argv:
	col = sys.argv[sys.argv.index('-col') + 1]
else:
	print("-col not given")
	sys.exit()

aseqs = seqdepotDB[col]

if '-id' in sys.argv:
	identifier = sys.argv[sys.argv.index('-id') + 1]
	cross = 1
else:
	cross = 0

if '--meta' in sys.argv:
	metaFlag = 1
else:
	metaFlag = 0



#identifier = sys.argv[2]
if not g_File:
	print("No input")
	sys.exit()

count = 0;
ptime = datetime.time

#print(ptime)

filein = open(g_File,"r");
for line in filein:
#	print(line.split('\t')[1])
	myjson = json.loads(line.split('\t')[1])
#print("hey");
#    my $son = substr($json,23);
#    print($son);
#    $json = $son;
    #my $aseq = from_json(substr($json,23));
    #print("$aseq->{_id}\n");
	count += 1
	foundAseq = aseqs.find_one({"_id":myjson["_id"]})
	if foundAseq:
		#print(foundAseq["_id"])
		addedDict = {}
		#if ((not metaFlag) and ('m' in foundAseq.keys()) and (foundAseq["m"]==1)):
		#	aseqs.update({"_id":myjson["_id"]},{"$set":{"m":0}})
		if cross == 1:
			theXfield = "x." + identifier
			for idElement in myjson["x"][identifier]:
				aseqs.update({"_id":myjson["_id"]},{"$addToSet":{theXfield:idElement}}) 
	else:
		print(count)
		#print(".")
		if metaFlag:
			myjson["m"] = 1
		else:
			myjson["m"] = 0
		#print(myjson)
		aseqs.insert(myjson)    

File: loadAseqsJson.pl
#!/usr/bin/perl

use strict;
use warnings;
use JSON;

use FindBin '$Bin';
use lib "$Bin";
use Common;


my $usage = <<USAGE;
Usage: $0 <database json>

  <database json> should consist of an entire json object per line

  Loads into the aseqs collection.

USAGE

my $g_File = shift or die $usage;
my $aseqs = &Common::aseqs();

#my $fh = &Common::openFileOrGzFile($g_File);

&Common::startTicker(10000);
my $count = 0;
open(FH,"|/usr/bin/tac $g_File");
while (<FH>) {
    my $json = $_;
#print("hey");
#    my $son = substr($json,23);
#    print($son);
#    $json = $son;
    #my $aseq = from_json(substr($json,23));
    my $aseq = from_json(substr($_,23));
    #print("$aseq->{_id}\n");
    if ($aseqs->find_one({_id => $aseq->{_id}})) {
        #print("$aseq->{_id}\n");
	
	#print($count++);
        # This assumes that the $aseq->{x}->{gi} is the most recent gi
	# list and will overwrite any pre-existing ones.
	#my %set = (
	#    'x.gi' => $aseq->{x}->{gi}
	#);
	#$aseqs->update({_id => $aseq->{_id}}, {'$set' => \%set});
	&Common::tick();
	next;
    }
    else {
    print("$count\t$aseq->{_id}\n________________________\n");
    $count++;
    #print($aseq); 
    my $hasUndoneData = index($aseq->{_s}, '-') != -1;
    if ($hasUndoneData) { $aseq->{$Common::paddingKey} =
    $Common::padding{basic}->{buffer}; }

    $aseqs->insert($aseq);
    &Common::removePadding($aseqs, $aseq->{_id});
    &Common::tick();
    }
}
